### PR TITLE
Some optimizations

### PR DESCRIPTION
**Pull request by bluerthanever**
For details please refer to each commits.

### CODE REVIEW

**Comment by bluerthanever** [comment]
I didn't put cases where a user uses proxies with domains instead of ips into consideration...
so if a user uses proxies with domains, the regular expression pattern need to be changes.

**Comment by bluerthanever** [comment]
I didn't put cases where a user uses proxies with domains instead of ips into consideration...
so if a user uses proxies with domains, the regular expression pattern needs to be changed.

File: PixivConfig.py [hunk 1]
```diff
@@ -206,12 +210,23 @@ class PixivConfig():
...
         for item in self.__items:
             setattr(self, item.option, item.process_value(item.default))
-        self.proxy = {'http': self.proxyAddress, 'https': self.proxyAddress}
+
+    @property
+    def proxy(self):
+        value = getattr(self, "proxyAddress", None)
+        if not value:
+            return None
+        match = re.match(r"^(?:(http|socks[45])://)?(\d+\.\d+\.\d+\.\d+):(\d+)$", value)
```
Comment (Nandaka) [inline PixivConfig.py:222]: maybe can use this pattern `^(https|http|socks[45])://[\w.]+(:?\d*)$`
image

Comment (Nandaka) [reply PixivConfig.py:222]: updated to allow `-` = `^(https|http|socks[45])://[\w.-]+(:?\d*)$`
image

**Pull request merged**